vm/src/builtins/type.rs

```diff
@@ -523,6 +523,57 @@ impl PyType {
         Ok(())
     }
 
+    #[pygetset(magic)]
+    fn annotations(&self, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        if !self.slots.flags.has_feature(PyTypeFlags::HEAPTYPE) {
+            return Err(vm.new_attribute_error(format!(
+                "type object '{}' has no attribute '__annotations__'",
+                self.name()
+            )));
+        }
+
+        Ok(self
+            .attributes
+            .read()
+            .get(identifier!(vm, __annotations__))
+            .cloned()
+            .unwrap_or_else(|| {
+                self.attributes
+                    .write()
+                    .insert(identifier!(vm, __annotations__), vm.ctx.new_dict().into())
+                    .unwrap_or_else(|| vm.ctx.new_dict().into())
+            }))
```

Comment:
Because `entry()` doesn't fit to rwlock very well, I suggest this form.
```suggestion
        let __annotations__ = identifier!(vm, __annotations__);
        let annotations = if let Some(annotations) = self.attributes.read().get(__annotations__) {
            annotations.clone()
        } else {
            let annotations: PyObjectRef = vm.ctx.new_dict().into();
            let removed = self
                .attributes
                .write()
                .insert(__annotations__, annotations.clone());
            debug_assert!(removed.is_none());
            annotations
        };
        Ok(annotations)
```

Reply:
Can I use `write()` in `else` while using `read()` in `if let`? I tried it, but an error seems to occur in this part, so I edited this part.